drivers/spi/spi_andes_atcspi200.c

```diff
@@ -162,31 +191,414 @@ static int configure(const struct device *dev,
+	remain_len = data->chunk_len - ctx->current_rx->len;
+	spi_context_update_rx(ctx, dfs, ctx->current_rx->len);
+
+	data->dma_rx.dma_blk_cfg.source_address = (uint32_t)SPI_DATA(cfg->base);
+
+	if (data->dma_rx.src_addr_increment) {
+		data->dma_rx.dma_blk_cfg.source_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+	} else {
+		data->dma_rx.dma_blk_cfg.source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+	}
+
+	data->dma_rx.dma_cfg.head_block = &data->dma_rx.dma_blk_cfg;
+	data->dma_rx.dma_cfg.head_block->next_block = NULL;
+	data->dma_rx.dma_cfg.user_data = data;
+
+	if (data->dma_rx.dma_cfg.source_chaining_en) {
+		data->dma_rx.dma_cfg.block_count = ctx->rx_count;
+		data->dma_rx.dma_cfg.dma_callback = NULL;
+		data->dma_rx.block_idx = 0;
+		struct dma_block_config *blk_cfg = &data->dma_rx.dma_blk_cfg;
+		const struct spi_buf *current_rx = ctx->current_rx;
+
+		while (remain_len > 0) {
+			struct dma_block_config *next_blk_cfg;
+
+			next_blk_cfg = &data->dma_rx.chain_block[data->dma_rx.block_idx];
+			data->dma_rx.block_idx += 1;
+
+			blk_cfg->next_block = next_blk_cfg;
+			current_rx = ctx->current_rx;
+
+			next_blk_cfg->block_size = current_rx->len /
+						data->dma_rx.dma_cfg.dest_data_size;
+
+			/* rx direction has periph as source and mem as dest. */
+			if (current_rx->buf == NULL) {
+				/* if rx buff is null, then write data to dummy address. */
+				next_blk_cfg->dest_address = (uintptr_t)&dummy_rx_tx_buffer;
+				next_blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+			} else {
+				next_blk_cfg->dest_address = (uintptr_t)current_rx->buf;
+				if (data->dma_rx.dst_addr_increment) {
+					next_blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+				} else {
+					next_blk_cfg->dest_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+				}
+			}
+
+			next_blk_cfg->source_address = (uint32_t)SPI_DATA(cfg->base);
+
+			if (data->dma_rx.src_addr_increment) {
+				next_blk_cfg->source_addr_adj = DMA_ADDR_ADJ_INCREMENT;
+			} else {
+				next_blk_cfg->source_addr_adj = DMA_ADDR_ADJ_NO_CHANGE;
+			}
+
+			blk_cfg = next_blk_cfg;
+			next_blk_cfg->next_block = NULL;
+			remain_len -= ctx->current_rx->len;
+			spi_context_update_rx(ctx, dfs, ctx->current_rx->len);
+		}
+	} else {
+		data->dma_rx.dma_blk_cfg.next_block = NULL;
+		data->dma_rx.dma_cfg.block_count = 1;
+		data->dma_rx.dma_cfg.dma_callback = dma_rx_callback;
+	}
+
+	/* pass our client origin to the dma: data->dma_rx.channel */
+	ret = dma_config(data->dma_rx.dma_dev, data->dma_rx.channel,
+			&data->dma_rx.dma_cfg);
+	/* the channel is the actual stream from 0 */
+	if (ret != 0) {
+		data->dma_rx.block_idx = 0;
+		data->dma_rx.dma_blk_cfg.next_block = NULL;
+		return ret;
+	}
+
+	return 0;
+}
+
+static int spi_transfer_dma(const struct device *dev)
+{
+	const struct spi_atcspi200_cfg * const cfg = dev->config;
+	struct spi_atcspi200_data * const data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	uint32_t data_len, tctrl, dma_rx_enable, dma_tx_enable;
+	int error = 0;
+
+	data_len = data->chunk_len - 1;
+
```

Comment:
you can remove that empty line as your are testing in the if below the variable you assigned previously (apply that were relevant)

Reply:
Ok, thanks a lot.